integration/spark/src/main/scala/org/apache/carbondata/indexserver/IndexJobs.scala

```diff
@@ -47,6 +47,9 @@ class DistributedIndexJob extends AbstractIndexJob {
 
   override def execute(indexFormat: IndexInputFormat,
       configuration: Configuration): util.List[ExtendedBlocklet] = {
+    // get only carbon segments.
+    indexFormat.setValidSegments(indexFormat.getValidSegments.asScala
```

Comment:
In Fall back mode, this scenario is handled?

Reply:
Yes, in the fallback mode it uses `getCarbonSegments` method.

Reply:
In EmbeddedIndexJob fall back mode, i think, you have to handle

Reply:
in EmbeddedIndexJob fallback mode, we have `toDistributable ` method, here `loadmetadatadetails `for segments are present and carbon segments are filtered.